x-pack/plugins/apm/server/routes/metrics.ts

```diff
@@ -18,21 +19,31 @@ const defaultErrorHandler = (err: Error) => {
 
 export function initMetricsApi(core: CoreSetup) {
   const { server } = core.http;
+
   server.route({
     method: 'GET',
     path: `/api/apm/services/{serviceName}/metrics/charts`,
     options: {
       validate: {
-        query: withDefaultValidators()
+        query: withDefaultValidators({
+          agentName: Joi.string().required()
+        })
       },
       tags: ['access:apm']
     },
     handler: async req => {
       const setup = setupRequest(req);
       const { serviceName } = req.params;
-      return await getAllMetricsChartData({
+      // ternary type guard required here because req.query can be type: string | RequestQuery
+      // and req.query.* can each have a value of type: string | string[]
+      const agentName =
+        typeof req.query !== 'string' && typeof req.query.agentName === 'string'
+          ? req.query.agentName
+          : '';
```

Comment:
Ahh, yes. There is this issue about it: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/25605.
I think the recommended approach around this is casting, which we also do in other places:
https://github.com/elastic/kibana/blob/dcf5f91784e320cd097d63b2df38622b03883e50/x-pack/plugins/apm/server/routes/errors.ts#L41-L44

With casting we can solve it like:
```tsx
const { agentName } = req.query as {
  agentName: string;
};
```

I think this is actually okay, because if `req.query.agentName` turns out not to be a string, we want it to blow up, and not just swallow the error. Ofc I'd prefer it to blow up at compile time instead of runtime but doesn't seem like we can get that.

Reply:
Works for me. Thanks!